lib/package/dynamodb/compileMethodsToDynamodb.js

```diff
@@ -0,0 +1,387 @@
+        },
+        {
+          StatusCode: 500,
+          SelectionPattern: '5\\d{2}',
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        }
+      ]
+    }
+
+    if (http && http.cors) {
+      let origin = http.cors.origin
+      if (http.cors.origins && http.cors.origins.length) {
+        origin = http.cors.origins.join(',')
+      }
+
+      integrationResponse.IntegrationResponses.forEach(async (val, i) => {
+        integrationResponse.IntegrationResponses[i].ResponseParameters = {
+          'method.response.header.Access-Control-Allow-Origin': `'${origin}'`
+        }
+      })
+    }
+
+    _.merge(integration, integrationResponse)
+
+    return {
+      Properties: {
+        Integration: integration
+      }
+    }
+  },
+
+  getDynamodbIntegrationRequestTemplates(http) {
+    const defaultRequestTemplates = this.buildDefaultDynamodbRequestTemplates(http)
+    return Object.assign(defaultRequestTemplates, _.get(http, ['request', 'template']))
+  },
+
+  buildDefaultDynamodbRequestTemplates(http) {
+    return {
+      'application/json': this.buildDefaultDynamodbRequestTemplate(http),
+      'application/x-www-form-urlencoded': this.buildDefaultDynamodbRequestTemplate(http)
+    }
+  },
+
+  getDynamodbObjectHashkeyParameter(http) {
+    if (http.hashKey.pathParam) {
+      return {
+        key: http.hashKey.pathParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().path.${http.hashKey.pathParam}`
+      }
+    }
+
+    if (http.hashKey.queryStringParam) {
+      return {
+        key: http.hashKey.queryStringParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().querystring.${http.hashKey.queryStringParam}`
+      }
+    }
+
+    if (http.hashKey.bodyParam) {
+      return {
+        key: http.hashKey.bodyParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$util.parseJson($input.body).${http.hashKey.bodyParam}`
+      }
+    }
+
+    return {
+      key: http.hashKey,
+      attributeType: 'S',
+      attributeValue: '$context.requestId'
+    }
+  },
+
+  getDynamodbObjectRangekeyParameter(http) {
+    if (http.rangeKey.pathParam) {
+      return {
+        key: http.rangeKey.pathParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$input.params().path.${http.rangeKey.pathParam}`
+      }
+    }
+
+    if (http.rangeKey.queryStringParam) {
+      return {
+        key: http.rangeKey.queryStringParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$input.params().querystring.${http.rangeKey.queryStringParam}`
+      }
+    }
+
+    if (http.rangeKey.bodyParam) {
+      return {
+        key: http.rangeKey.bodyParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$util.parseJson($input.body).${http.rangeKey.bodyParam}`
+      }
+    }
+
+    return {
+      key: http.rangeKey,
```

Comment:
this seems to contradict with the validate.test where it says it should throw if we give a `rangeKey` when the `method` is post and there's no `action`